include/mfmg/sparse_matrix_device.templates.cuh

```diff
@@ -266,6 +303,7 @@ void SparseMatrixDevice<ScalarType>::mmult(
   // TODO Communicate the values
 
   // Compute the number of non-zero elements in C
+  ASSERT(B.m() == n(), "The matrices cannot be multiplied together");
```

Comment:
Improve the error message by also printing the sizes of matrices you are trying to multiply.